Entities/Industry/Fireplace/Fireplace.as

```diff
@@ -116,9 +121,13 @@ void Ignite(CBlob@ this)
 
 f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitterBlob, u8 customData)
 {
-	if (customData == Hitters::water)
+	if (isWaterHitter(customData)) 
 	{
 		Extinguish(this);
 	}
+	if (isIgniteHitter(customData))
```

Comment:
Use `else if` since logically something cannot both extinguish and ignite a fireplace.